drivers/char/broadcom/vc_sm/vmcs_sm.c

```diff
@@ -276,7 +276,10 @@ static unsigned int vmcs_sm_vc_handle_from_pid_and_address(unsigned int pid,
 	/* Lookup the resource. */
 	if (!list_empty(&sm_state->map_list)) {
 		list_for_each_entry(map, &sm_state->map_list, map_list) {
-			if (map->res_pid != pid || map->res_addr != addr)
+			if (map->res_pid != pid)
+				continue;
+			if (!(map->res_addr <= addr &&
```

Comment:
Yeah, mine was obscure and not efficient. Fixed in https://github.com/raspberrypi/linux/pull/2333/commits/872f3ca2c4dd268bc6d537151435e45f16224b0e.

Reply:
(I changed the second condition of yours from `addr > (map->res_addr + map->resource->res_size)` to `addr >= (map->res_addr + map->resource->res_size)` because yours may match an neighboring map area.)